src/inference/src/dev/threading/cpu_streams_executor.cpp

```diff
@@ -335,27 +334,72 @@ struct CPUStreamsExecutor::Impl {
     // will be counted by thread_local t_stream_count_map.
     // when the customer's thread is destoryed, the stream's count will became 1,
     // Call local() will reuse one of them, and release others.
+    // it's only a workaround for ticket CVS-111490, please be carefully when need to modify
+    // CustomeThreadLocal::local(), especially like operations that will affect the count of
+    // CustomThreadLocal::ThreadId
     class CustomThreadLocal : public ThreadLocal<std::shared_ptr<Stream>> {
+        class ThreadTracker {
+        public:
+            explicit ThreadTracker(const std::thread::id& id)
+                : _id(id),
+                  _count_ptr(std::make_shared<std::atomic_int>(1)) {}
+            ~ThreadTracker() {
+                 _count_ptr->fetch_sub(1);
+            }
+            std::shared_ptr<ThreadTracker> fetch() {
+                ThreadTracker* new_ptr = new ThreadTracker(*this);
```

Comment:
```suggestion
                auto new_ptr = std::make_shared<ThreadTracker>(*this);
```

Reply:
I tried it but it show compile error, the compile said it's a private function

Reply:
fix it, create the shared_ptr before assert